File: src/main/java/com/morph/engine/graphics/FontManager.java
package com.morph.engine.graphics;

import java.util.HashMap;

/**
 * Created on 7/31/2017.
 */
public class FontManager {
    private static HashMap<String, FontResource> loadedFonts = new HashMap<>();

    public static FontResource loadFont(String font) {
        if (loadedFonts.containsKey(font)) return loadedFonts.get(font);

        FontResource newFont = new FontResource(new LoadedFont(font));
        loadedFonts.put(font, newFont);
        return newFont;
    }

    public static void unloadFont(String font) {
        if (loadedFonts.get(font).removeReference()) {
            loadedFonts.remove(font);
        }
    }
}

File: src/main/kotlin/com/morph/engine/graphics/FontManager.kt
package com.morph.engine.graphics

import java.util.*

/**
 * Created on 7/31/2017.
 */
object FontManager {
    private val loadedFonts = HashMap<String, FontResource>()

    fun loadFont(font: String) = loadedFonts[font] ?: FontResource(LoadedFont(font)).apply { loadedFonts[font] = this }

    fun unloadFont(font: String) = loadedFonts[font]?.let {
        if (it.removeReference()) loadedFonts.remove(font)
    }
}
